src/oneshot/source/oneshot.cpp

```diff
@@ -570,6 +597,199 @@ bool Oneshot::msgbox(int type, const char *body, const char *title)
 #endif // #ifdef OS_LINUX
 }
 
+bool Oneshot::addNotifyIcon(const char* tip)
+{
+	if (p->trayIcon) {
+		Debug() << "Already added tray icon";
+		return false;
+	}
+
+#ifdef OS_W32
```

Comment:
Please move this code to two different files instead of an #ifdef where possible, then include the file from meson depending on the operating system